packages/server/src/conversations/api.ts

```diff
@@ -58,6 +58,11 @@ export class ConversationApi {
   async recent(req: ClientApiRequest, res: Response) {
     const userId = req.params.userId as uuid
 
+    const user = await this.users.get(userId)
+    if (!user || user.clientId !== req.client.id) {
+      return res.sendStatus(404)
+    }
```

Comment:
Otherwise, it would throw 500 HTTP errors since we might try to create a conversation with a userId that does not exist. This would cause the database to send foreign key constrain violations. I copied this piece of code from the list function above.